expyfun/tests/test_version.py

```diff
@@ -24,35 +21,45 @@ def test_version():
         assert_version(__version__[-7:])
     assert_true(all('actual' in str(ww.message) for ww in w))
 
-    v = '090948e'
-    if not _has_git:
-        assert_raises(ImportError, download_version, v, tempdir)
-    else:
-        assert_raises(IOError, download_version, v, op.join(tempdir, 'foo'))
-        assert_raises(RuntimeError, download_version, 'x' * 7, tempdir)
-        download_version(v, tempdir)
-        ex_dir = op.join(tempdir, 'expyfun')
-        assert_true(op.isdir(ex_dir))
-        assert_true(op.isfile(op.join(ex_dir, '__init__.py')))
-        with open(op.join(ex_dir, '_version.py')) as fid:
-            line1 = fid.readline().strip()
-        assert_equal(line1.split(' = ')[1][-8:-1], v)
-
-        # auto dir determination
-        orig_dir = os.getcwd()
-        os.chdir(tempdir)
-        try:
-            assert_true(op.isdir('expyfun'))
-            assert_raises(IOError, download_version, v)
-        finally:
-            os.chdir(orig_dir)
-        # make sure we can get latest version
-        download_version(dest_dir=tempdir_2)
+    for want_version in ('090948e', 'cae6bc3', 'b6e8a81'):  # old, broken, new
+        tempdir = _TempDir()
+        tempdir_2 = _TempDir()
+        if not _has_git:
+            assert_raises(ImportError, download_version, want_version, tempdir)
+        else:
+            assert_raises(IOError, download_version, want_version,
+                          op.join(tempdir, 'foo'))
+            assert_raises(RuntimeError, download_version, 'x' * 7, tempdir)
```

Comment:
I'm not sure how that's possible, since new `_TempDir()`s are getting created each time.  Can you confirm?